packages/client/src/Record.ts

```diff
@@ -76,6 +84,23 @@ export class CollectionRecord<T> {
   })
 }
 
+export class CollectionRecordResponse<T> extends CollectionRecord<T> {
+  data: T | null
+  block: {
+    hash: string
+  } | null
+
+  constructor(id: string, data: T | null, collection: Collection<T>, client: Client, onSnapshotRegister: CollectionRecordSnapshotRegister<T>) {
+    super(id, collection, client, onSnapshotRegister)
+    this.data = data
+    this.block = null
+  }
+
+  exists = (): boolean => {
```

Comment:
Unless we can make this a type guard, I think it might be better to remove it and users can instead use `if (record.data)`, such that Typescript knows if record.data is null or not